images/installer/root/usr/local/bin/generate

```diff
@@ -0,0 +1,347 @@
+
+    master_public_url = y.get("masterPublicURL")
+    if not master_public_url:
+        print "'kubernetesMasterConfig.masterPublicURL' missing from '{}'; Unable to connect to master host...".format(MASTER_CONFIG)
+        exit(1)
+
+    # connect to remote host using `oc login...` and extract all possible node information
+    oc = OpenShiftClient()
+    oc.login(master_public_url, openshift_cluster_user, openshift_cluster_pass)
+    nodes_config = yaml.load(oc.get_nodes())
+
+    # contains host types (e.g. masters, nodes, etcd)
+    host_groups = dict()
+    openshift_hosted_logging_deploy = False
+    ansible_become = "yes"
+    is_etcd_deployed = master_config.get("storage-backend", "") in ["etcd3", "etcd2", "etcd"]
+
+    if asset_config and asset_config.get('loggingPublicURL'):
+        openshift_hosted_logging_deploy = True
+
+    m = Host("masters")
+    m.host_alias(common_host_alias)
+    m.address(master_config["masterIP"])
+    m.public_host_name(master_public_url)
+    host_groups["masters"] = HostGroup([m])
+
+    if nodes_config:
+        node_hosts = list()
+        for node in nodes_config.get("items", []):
+            if node["kind"] != "Node":
+                continue
+
+            n = Host("nodes")
+            n.host_alias(common_host_alias)
+
+            address = ""
+            internal_hostname = ""
+            for item in node["status"].get("addresses", []):
+                if not address and item['type'] in ['InternalIP', 'LegacyHostIP']:
+                    address = item['address']
+
+                if item['type'] == 'Hostname':
+                    internal_hostname = item['address']
+
+            n.address(address)
+            n.host_name(internal_hostname)
+            node_hosts.append(n)
+
+        host_groups["nodes"] = HostGroup(node_hosts)
+
+    if etcd_config:
+        etcd_hosts = list()
+        for url in etcd_config.get("urls", []):
+            e = Host("etcd")
+            e.host_alias(common_host_alias)
+            e.host_name(url)
+            etcd_hosts.append(e)
+
+        host_groups["etcd"] = HostGroup(etcd_hosts)
+
+    # open new inventory file for writing
+    try:
+        inv_file_obj = open(HOME + '/' + INVY, 'w+')
+    except IOError as err:
+        print "Unable to create or open generated inventory file: {}".format(err)
+        exit(1)
+
+    inv_file_obj.write("[OSEv3:children]\n")
+    for group in host_groups:
+        inv_file_obj.write("{}\n".format(group))
+    inv_file_obj.write("\n")
+
+    inv_file_obj.write("[OSEv3:vars]\n")
+    inv_file_obj.write("ansible_ssh_user={}\n".format(ansible_ssh_user))
+    inv_file_obj.write("ansible_become={}\n".format(ansible_become))
+    inv_file_obj.write("openshift_uninstall_images={}\n".format(str(openshift_uninstall_images)))
+    inv_file_obj.write("openshift_deployment_type={}\n".format(openshift_deployment_type))
+    inv_file_obj.write("openshift_install_examples={}\n".format(str(openshift_install_examples)))
```

Comment:
What about all the other stuff from my config file?

```
openshift_release: 3.6
openshift_image_tag: v3.6.0
openshift_hosted_logging_deploy: True
openshift_logging_image_version: v3.6.0
openshift_disable_check: disk_availability,docker_storage,memory_availability
```

etc.... i think we want to be able to generate as close to the original file as possible.